Macros/DMPED_pipeline_read_update_file.sas

```diff
@@ -0,0 +1,375 @@
+	units_51_60_AMI = "Number of units at 51-60% of AMI"
+	units_61_80_AMI = "Number of units at 61-80% of AMI"
+	mar_ward = "MAR Ward"
+	inclusionary_zoning = "Property is inclusionary zoning"
+	agency = "DC Government Agency"
+	PUD = "Property is part of a PUD"
+	Start_closing_date = "Closing date"
+	Estimated_or_Actual_start = "Actual or estimated start date?"
+	Construction_End_Date = "Construction completion date"
+	Estimated_or_Actual_End_Date = "Actual or estimated end date?"
+	Address_Postal_Code = "Zip code"
+	ADU = "Project is an ADU"
+	Affordable_units_manual_entry = "Manual entry of number of units"
+	DCHA_Units_0_30_AMI = "Number of DCHA units at 0-30% of AMI"
+	DCHA_Units_31_50_AMI = "Number of DCHA units at 31-50% of AMI"
+	DCHA_Units_51_60_AMI = "Number of DCHA units at 51-60% of AMI"
+	DCHA_Units_61_80_AMI = "Number of DCHA units at 61-80% of AMI"
+	DCHA_Units_Affordable_total = "Total number of DCHA affordable units"
+	DCHFA_UnitsUnits_0_30_AMI = "Number of DCHFA units at 0-30% of AMI"
+	DCHFA_Units_31_50_AMI = "Number of DCHFA units at 31-50% of AMI"
+	DCHFA_Units_51_60_AMI = "Number of DCHFA units at 51-60% of AMI"
+	DCHFA_Units_61_80_AMI = "Number of DCHFA units at 61-80% of AMI"
+	DCHFA_Units_Affordable = "Total number of DCHFA affordable units"
+	DHCD_DFD_0_30_AMI = "Number of DHCD-DFD units at 0-30% of AMI"
+	DHCD_DFD_31_40_AMI = "Number of DHCD-DFD units at 31-40% of AMI"
+	DHCD_DFD_41_50_AMI = "Number of DHCD-DFD units at 41-50% of AMI"
+	DHCD_DFD_51_60_AMI = "Number of DHCD-DFD units at 51-60% of AMI"
+	DHCD_DFD_61_80_AMI = "Number of DHCD-DFD units at 61-80% of AMI"
+	DHCD_DFD_allsubsidy = "Total number of DHCD-DFD subsidized units"
+	DHCD_DFD_pubhousing = "Total number DHCD-DFD public housing units"
+	DHCD_DFD_total = "Total number of DHCD-DFD affordable units"
+	DMPED_PPD_0_30_AMI ="Number of DMPED-PPD units at 0-30% of AMI"
+	DMPED_PPD_31_50_AMI ="Number of DMPED-PPD units at 31-50% of AMI"
+	DMPED_PPD_51_60_AMI ="Number of DMPED-PPD units at 51-60% of AMI"
+	DMPED_PPD_61_80_AMI ="Number of DMPED-PPD units at 61-80% of AMI"
+	HPTF_Affordable_Units = "Number of Housing Production Trust Fund affordable units"
+	IZ_Database_total = "Total number of IZ units"
+	IZ_Database_0_50_AMI = "Number of IZ units at 0-50% of AMI"
+	IZ_Database_51_60_AMI = "Number of IZ units at 51-60% of AMI"
+	IZ_Database_61_80_AMI = "Number of IZ units at 61-80% of AMI"
+	PUD_Affordable_Units = "Total number of PUD affordable units"
+	PUD_Units_0_30_MFI = "Number of PUD units at 0-30% of MFI"
+	PUD_Units_31_60_MFI = "Number of PUD units at 31-60% of MFI"
+	PUD_Units_61_80_MFI = "Number of PUD units at 61-80% of MFI"
+	PUD_Units_81_MFI = "Number of PUD units over 81% of MFI"
+	TOPA = "TOPA Project"
+	DHCD_0_30_AMI = "Number of DHCD units at 0-30% of AMI"
+	DHCD_31_50_AMI = "Number of DHCD units at 31-50% of AMI"
+	DHCD_51_60_AMI = "Number of DHCD units at 51-60% of AMI"
+	DHCD_61_80_AMI = "Number of DHCD-DFD units at 61-80% of AMI";
+
+  run;
```

Comment:
For all three data sets, it's helpful if we convert some of the text-based fields, like _Status_ and _Estimated_or_Actual_start_, to coded values with formats. In a separate program (I usually call it **Make_formats.sas**), you can define these formats and save them in the DMPED library. As an example, you can look at the [Police/Prog/Make_formats.sas](https://github.com/NeighborhoodInfoDC/Police/blob/master/Prog/Make_formats.sas) program. Note that you need to create these as _permanent_ formats by including the library name in the _Proc Format_ statement. Since these are permanent formats saved to the library, you have to batch submit **Make_formats.sas** on L: before the formats will be created. 

When creating the data sets, you would use _if/then/else_ or [_select/when_ statements](https://blogs.sas.com/content/iml/2016/06/20/select-when-sas-data-step.html), to create a new version of variables like _Status_ that are coded "1" for completed, "2" for under construction, etc. Then assign the format to the new _Status_ variable using a _format_ statement in a data step. 

The result of this will be that the individual formatted values will be listed in the metadata, which will make it easier for someone who is using the file.

For the yes/no fields (_Inclusionary_Zoning_, _ADU_, etc.) I would recode them to numeric 0/1 dummy variables (1="yes", 0="no"). This will allow you to use expressions like `if Inclusionary_Zoning then ..`, since SAS interprets 1 as "true," and 0 as "false." You should assign these 0/1 variables the _dyesno._ format, which is a standard format in our General library (i.e., you don't need to create it). And, again, the separate values will be listed in the metadata. 

Let me know if you have any questions about that.